### ISSUE TITLE
[Bug]: Search work without file extension, but with it's not

### ISSUE THREAD

**Issue by TheAwiteb** [ParthJadhav/rust_search#4]
^ Title
---
For example i have file it's name is `some_random_word_yes_it_is.md` in `/home/User/foo/bar/`
```rust
use rust_search::Search;

fn main(){
    let search = Search::new(
        "/home/User/foo/bar/",
        Some("yes"),
        None,
        None);

    for path in search {
        println!("{}", path);
    }
}
```
This will work, but when i add the file extinction it's not, for example

```rust
use rust_search::Search;

fn main(){
    let search = Search::new(
        "/home/User/foo/bar/",
        Some("yes"),
        Some(".md"), // also "md"
        None);

    for path in search {
        println!("{}", path);
    }
}
```

**Comment by ParthJadhav** [comment]
This needs to be fixed... Adding extension will require adding the exact file name. This shouldn't be the case. 

Will work on a fix. Thanks for reporting

**Comment by TheAwiteb** [comment]
I worked around it that way,`Some(r".*\.md")` but it's will be nice to be builtin

**Comment by ParthJadhav** [comment]
Hey @TheAwiteb , Thanks for the suggestion. Implemented it in this PR https://github.com/ParthJadhav/rust_search/pull/5 . Can you test it and review the code so we can merge it...

**Comment by TheAwiteb** [comment]
@ParthJadhav I think this was fixed by #9

**Comment by TheAwiteb** [comment]
Fixed by #9
